cocos/renderer/gfx-gles2/GLES2GPUObjects.h

```diff
@@ -86,22 +86,60 @@ struct GLES2GPUInput {
 };
 typedef vector<GLES2GPUInput>::type GLES2GPUInputList;
 
-struct GLES2GPUUniform {
-  uint binding;
-  String name;
-  GFXType type;
-  uint stride;
-  uint count;
-  uint size;
-  uint offset;
-  GLenum gl_type;
-  GLint gl_loc;
-  uint8_t* buff = nullptr;
+struct GLES2GPUUniform
+{
+    uint binding;
+    String name;
+    GFXType type;
+    uint stride;
+    uint count;
+    uint size;
+    uint offset;
+    GLenum gl_type;
+    GLint gl_loc;
+    uint8_t* buff = nullptr;
+
+    GLES2GPUUniform() {}
+    GLES2GPUUniform(const GLES2GPUUniform& rhs)
+    :binding(rhs.binding),
+    name(rhs.name),
+    type(rhs.type),
+    stride(rhs.stride),
+    count(rhs.count),
+    size(rhs.size),
+    offset(rhs.offset),
+    gl_type(rhs.gl_type),
+    gl_loc(rhs.gl_loc)
+    {
+        CC_SAFE_FREE(buff);
+        buff = (uint8_t*)CC_MALLOC(size);
+        memcpy(buff, rhs.buff, size);
+    }
 
-  GLES2GPUUniform() {}
-  ~GLES2GPUUniform() {
-    CC_SAFE_FREE(buff);
-  }
+    GLES2GPUUniform& operator =(const GLES2GPUUniform& rhs)
+    {
+        if(this != &rhs)
+        {
+            binding = rhs.binding;
+            name = rhs.name;
+            type = rhs.type;
+            stride = rhs.stride;
+            count = rhs.count;
+            size = rhs.size;
+            offset = rhs.offset;
+            gl_type = rhs.gl_type;
+            gl_loc = rhs.gl_loc;
+            CC_SAFE_FREE(buff);
```

Comment:
use CC_FREE and CC_NEW instead to keep the same memory mechanism.

Reply:
CC_NEW can not allocate specific bytes, it was used to new an Object.